#460 - Added the Image Info annotation to the Peregrine Sling Module

reggie7 opened the pull request:
Implements #460.
Replaces #462.
Base for [percli#24](https://github.com/headwirecom/percli/pull/24) and [themeclean-flex#175](https://github.com/headwirecom/themeclean-flex/pull/175).

Commit:
#460 - Added the Image Info annotation to the Peregrine Sling Module

Commit:
Merge branch 'develop' into issues/460-develop

Commit:
Removed per-model and moved the code into base.core

Commit:
Removed the per-model module

Commit:
Added the Image Info to the Themeclean’s Media Block Model

reggie7 committed:
Merge branch 'develop' into issues/460-develop

Commit:
Merge branch 'develop-sling12' into issues/460-develop-sling12

# Conflicts:
#	platform/base/core/pom.xml
#      Fix failing tests

Commit:
Merge branch 'issues/460-develop' into issues/460-develop-sling12

Commit:
Remove unused stuff

Commit:
Grab dimensions from the source

reusr1 commented on platform/base/core/src/main/java/com/peregrine/model/impl/ImageInfoInjector.java:
```diff
@@ -0,0 +1,117 @@
+            return null;
+        }
+
+        final BufferedImage image = getImage(resource, ((ImageInfo) annotation).name());
+        if (isNull(image)) {
+            return null;
+        }
+
+        return new Dimension(image.getWidth(), image.getHeight());
```
I am not sure if I like this approach - we fetch the image and parse it - that is a lot of computational overhead for each image served. We already extract the image width and height into the meta data at upload of the image. Would it not make more sense to use the data from there?  @schaefa @reggie7 ?

reggie7 replied:
Yup, it does, I have actually found it during my work on #17, adjusted and just pushed the code a couple of minutes ago.

Commit:
Extract PerAsset.getOrSaveAndGetDimension()

cmrockwell committed:
added sling junit tests for perAssets

cmrockwell committed:
added tests and test images for example site. SVG should be supported, but is not yet.

Commit:
Merge branch 'develop-sling12' into issues/460-develop-sling12

reggie7 committed:
Merge pull request #676 from headwirecom/issues/460-develop-sling12-junit

Issues/460 develop sling12 junit

Commit:
Merge branch 'issues/460-develop-sling12' of https://github.com/headwirecom/peregrine-cms into issues/460-develop-sling12

Commit:
Merge branch 'develop-sling12' into issues/460-develop-sling12

Commit:
Cover SVG case

We use direct approach instead of a library since it's faster to implement at this point.
Let's go through all the hassle to enlist a bigger lib like batik when there's actually a justification
for such effort (apparently unfortunately we don't yet have a nice and quick ootb dependency bundles
inclusion mechanism).

Commit:
Fix missing property to save the cards change

Commit:
Merge branch 'develop-sling12' into issues/460-develop-sling12

Commit:
Use try-with-resources to auto-close InputStream
